clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3253,10 +3253,24 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &AL) {
 
   ASTContext &Ctx = S.getASTContext();
 
+  // The arguments to reqd_work_group_size are ordered based on which index
+  // increments the fastest. In OpenCL, the first argument is the index that
+  // increments the fastest, and in SYCL, the last argument is the index that
+  // increments the fastest.
+  //
+  // The arguments to max_work_group_size are ordered based on which index
+  // increments the fastest. In SYCL, the last argument is the index that
+  // increments the fastest.
   if (const auto *A = D->getAttr<SYCLIntelMaxWorkGroupSizeAttr>()) {
-    if (!((getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal()) &&
-          (getExprValue(AL.getArgAsExpr(1), Ctx) <= *A->getYDimVal()) &&
-          (getExprValue(AL.getArgAsExpr(2), Ctx) <= *A->getZDimVal()))) {
+    bool checkFirstArgument = S.getLangOpts().OpenCL
+                              ? getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal()
```

Comment:
Is this correct? For OpenCL mode, Shouldn't this be  - 
`AL.getArgAsExpr(0), Ctx) <= *A->getZDimVal()`

I also feel its more readable when you check each argument of AL i.e. - 

```
bool checkFirstArgument = S.getLangOpts().OpenCL
                              ? getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getZDimVal()
			      : getExprValue(AL.getArgAsExpr(0), Ctx) <= *A->getXDimVal();
bool checkSecondArgument = getExprValue(AL.getArgAsExpr(1), Ctx) <= *A->getYDimVal();
bool checkThirdArgument = S.getLangOpts().OpenCL
                              ? getExprValue(AL.getArgAsExpr(2), Ctx) <= *A->getXDimVal()
			      : getExprValue(AL.getArgAsExpr(2), Ctx) <= *A->getZDimVal();

```